src/components/tabs/tabs.ts

```diff
@@ -0,0 +1,418 @@
+            ({ start, end }) => start <= point && end >= point
+          )
+        : findLastIndex(dimensions, ({ start }) => start <= point);
+    } else {
+      target = isNext
+        ? dimensions.findIndex(
+            ({ start, end }) => start <= point && end >= point
+          )
+        : findLastIndex(dimensions, ({ end }) => end >= point);
+    }
+    return this.tabs[target];
+  }
+
+  protected handleStartButtonClick() {
+    this.scrollToTab(this.getTabAtBoundary('start'), 'start');
+  }
+
+  protected handleEndButtonClick() {
+    this.scrollToTab(this.getTabAtBoundary('end'), 'end');
+  }
+
+  protected handleClick(event: MouseEvent) {
+    const target = event.target as HTMLElement;
+    const tab = target.closest('igc-tab');
+
+    if (!tab || tab.disabled) {
+      return;
+    }
+
+    tab.focus();
+    this.setSelectedTab(tab);
+    this.emitEvent('igcChange', { detail: this._selected });
+  }
+
+  protected handleKeyDown = (event: KeyboardEvent) => {
+    const { key } = event;
+    const enabledTabs = this.enabledTabs;
+
+    let index = enabledTabs.indexOf(document.activeElement as IgcTabComponent);
+
+    switch (key) {
+      case 'ArrowLeft':
+        index = this.isLTR
+          ? (enabledTabs.length + index - 1) % enabledTabs.length
+          : (index + 1) % enabledTabs.length;
+        break;
+      case 'ArrowRight':
+        index = this.isLTR
+          ? (index + 1) % enabledTabs.length
+          : (enabledTabs.length + index - 1) % enabledTabs.length;
+        break;
+      case 'Home':
+        index = 0;
+        break;
+      case 'End':
+        index = enabledTabs.length - 1;
+        break;
+      case 'Enter':
+      case ' ':
+        this.setSelectedTab(enabledTabs[index]);
+        break;
+      default:
+        return;
+    }
+
+    enabledTabs[index].focus({ preventScroll: true });
+
+    if (this.activation === 'auto') {
+      this.setSelectedTab(enabledTabs[index]);
+      this.emitEvent('igcChange', { detail: this._selected });
+    } else {
+      this.scrollToTab(enabledTabs[index]);
+    }
+
+    event.preventDefault();
+  };
+
+  private setAriaAttributes() {
+    this.tabs.forEach((tab) => {
+      const panel = this.panels.find((el) => el.id === tab.panel);
+      if (panel) {
+        tab.setAttribute('aria-controls', panel.id);
+        panel.setAttribute('aria-labelledby', tab.getAttribute('id')!);
+      }
+    });
+  }
+
+  @eventOptions({ passive: true })
+  protected handleScroll() {
+    this.updateScrollButtons();
+  }
+
+  protected handleSlotChange = () => {
+    this.resizeObserver?.disconnect();
+
+    if (!this.selectedTab) {
+      this.tabs.forEach((tab) => (tab.selected = false));
+      this.panels.forEach((panel) => (panel.style.display = 'none'));
+      this._selected = '';
+    }
+
+    this.setupObserver();
+  };
+
+  /** Selects the specified tab and displays the corresponding panel.  */
+  public select(selectedTab: string | HTMLElement) {
```

Comment:
Still want to perhaps drop the union here, however I checked the generated output and at least for now it picks the string only, so that's perhaps fine